### ISSUE TITLE
Build the Commons-Simulator cadCAD model so that it uses the CV, Hatch and ABC sims that we have

### ISSUE THREAD

**Issue by GriffGreen** [commons-stack/commons-simulator#4]
Ok so more background here:
Here are the simulations I am talking about:

Hatch Simulation: https://github.com/commons-stack/conviction/blob/master/hatch_sim.ipynb

Augmented Bonding Curve Simulation: https://github.com/commons-stack/conviction/blob/master/abc_sim.ipynb

Conviction Voting Simulation: https://github.com/commons-stack/conviction/blob/master/conviction_cadCAD3.ipynb

We should Fork these Sims and make them fit our use case...

1. Trim them down (remove all the plots etc) and optimize them for our use.

2. We should improve the agent based modeling assumptions as we play with them trying to make the agents match our assumptions of the RadX community that will use this.

3. Develop a system for the Sims to be able to take variables from the Front End (Probably done already)

4. We will need to make sure the sims can process the responses from the user and apply them to the Sim. For instance... when they answer the vesting question, they are specifying the Cliff and the 1/2life curve with one number... we will need to take that one number and have a simple equation that kicks out the relevant constants to cadCAD... something similar will need to be done for a few variables.

To be complete, these are the questions asked in the narrative which define our variables:

A. How many Hatchers you will include when initiating the first RadicalxChange Commons
B. How many proposals do you expect your band of Hatchers to be able to gather within the first month?

C. (Hatch Tribute) What % of our Hatchers contributions will be allocated to the Community Funding Pool, as opposed to collateralizing our local economy?
D. (Cliff + Vesting Curve 1/2 life) How many days will pass before 80% of the Hatcher’s tokens are unlocked?

E. (Exit Tribute) What percent of the released Reserve will go to the Funding Pool when tokens are sold?
F. (Kappa) What shape of the curve will the RxC Community choose for their economy?

G. (Alpha) How many days does it take to reach 80% of a voter’s max voting weight?
H. (Beta) What is the max percentage of the Funding Pool that can be spent in one proposal?

See https://docs.google.com/document/d/10eQzJ1XkvN66ltpRXq_vo8rxtDgarX5Q7C8TdJUlGD4/edit?usp=sharing
for more context.

**Comment by GriffGreen** [comment]
Ok so more background here: 
Here are the simulations I am talking about:

Hatch Simulation: https://github.com/commons-stack/conviction/blob/master/hatch_sim.ipynb

Augmented Bonding Curve Simulation: https://github.com/commons-stack/conviction/blob/master/abc_sim.ipynb

Conviction Voting Simulation: https://github.com/commons-stack/conviction/blob/master/conviction_cadCAD3.ipynb

We should Fork these Sims and make them fit our use case...

1. Trim them down (remove all the plots etc) and optimize them for our use. 

2. We should improve them as we play with them trying to make the agents match our assumptions of the RadX community that will use this.

3. Develop a system for the Sims can take variables from the Front End

4. We will need to make sure the sims can take the responces from the user and apply them to the Sim.
Specifically, these are the questions asked in the narrative:

1. **How many Hatchers** you will include when initiating the first RadicalxChange Commons
2.  **How many proposals** do you expect your band of Hatchers to be able to gather **within the first month**?

3. **(Hatch Tribute)** What % of our Hatchers contributions will be allocated to the Community Funding Pool, as opposed to collateralizing our local economy?
4. **(Cliff + Vesting Curve 1/2 life)** How many days will pass before 80% of the Hatcher’s tokens are unlocked?

5. **(Exit Tribute)** What percent of the released Reserve will go to the Funding Pool when tokens are sold? 
6. **(Kappa)** What shape of the curve will the RxC Community choose for their economy?

7. **(Alpha)** How many days does it take to reach 80% of a voter’s max voting weight?
8. **(Beta)** What is the max percentage of the Funding Pool that can be spent in one proposal?

See https://docs.google.com/document/d/10eQzJ1XkvN66ltpRXq_vo8rxtDgarX5Q7C8TdJUlGD4/edit?usp=sharing
 for more context.

**Comment by GriffGreen** [comment]
We already begun to integrate the hatch in the simulation in https://github.com/commons-stack/commons-simulator/issues/15

**Comment by randomshinichi** [comment]
A, B, C, D: don't seem to match anything in hatch_sim.ipynb. G, H seem to have to do with the Conviction Voting.

Augmented Bonding Curve: includes reserve (backs tokens), funding pool (funds things), curve that determines price

Hatch: includes vesting schedule... and not much else?

Conviction Voting: [the only good explanation available](https://medium.com/commonsstack/conviction-voting-a-novel-continuous-decision-making-alternative-to-governance-62e215ad2b3d) 
Basically, the longer you hold a position on something (and you can hold a position on multiple things), the longer the weight accumulates. Also, last minute vote switching doesn't work because the weight given to your previous preference drains gradually, doesn't disappear straight away.
It seems the only source of actual equations for this is Conviction Voting Module Google Doc? But it's not very understandable either...

**Comment by GriffGreen** [comment]
You are right that almost all the action is in the CV3 sim... thats the only one really doing the Agent based simulation... We need to tell cadCAD how many agents there are and how many proposals there are for the Agents to interact with... and that is happening in CV3

A: how many hatchers... how many people have the OPTION of buying into the Hatch: Modeling is in CV3

B: how many proposals in the first month... The Hatchers will have a lot of tokens and we need to figure out which of these proposals will get passed... Both of these have more modeling done in the CV3 model

C is theta in the Hatch sim

D is not modeled anywhere, we have to add that in to the sim

Conviction Voting resources are many... but its definitely not very well organized... Check the under the Table of Contents, under the heading "Resources" in our weekly sync doc: 
https://docs.google.com/document/d/1B6Tg-LPD538-AOsPeUAuM-_ER8VBcnmtbe5dZgaRvrY/edit#
